File: Windows/Services/SecurityPolicyService.cs
using System;
using System.Security;

namespace StartupDashboard.Services
{
    public interface ISecurityPolicyService
    {
        bool IsLockedOut { get; }
        void RecordFailedAttempt();
        void ResetAttempts();
        void ClearSensitiveData(SecureString sensitiveData);
    }

    public class SecurityPolicyService : ISecurityPolicyService
    {
        private int _failedAttempts = 0;
        private const int MaxAttempts = 5;
        private DateTime _lockoutExpiry = DateTime.MinValue;

        public bool IsLockedOut => DateTime.Now < _lockoutExpiry;

        public void RecordFailedAttempt()
        {
            _failedAttempts++;
            if (_failedAttempts >= MaxAttempts)
            {
                _lockoutExpiry = DateTime.Now.AddMinutes(15);
            }
        }

        public void ResetAttempts()
        {
            _failedAttempts = 0;
            _lockoutExpiry = DateTime.MinValue;
        }

        public void ClearSensitiveData(SecureString sensitiveData)
        {
            sensitiveData?.Dispose();
        }
    }
}

File: macOS/Services/SecurityPolicyService.swift
import Foundation

protocol SecurityPolicyServiceProtocol {
    var isLockedOut: Bool { get }
    func recordFailedAttempt()
    func resetAttempts()
}

class SecurityPolicyService: SecurityPolicyServiceProtocol {
    private(set) var failedAttempts = 0
    private let maxAttempts = 5
    private(set) var lockoutExpiry = Date.distantPast

    var isLockedOut: Bool {
        return Date() < lockoutExpiry
    }

    var attemptsRemaining: Int {
        return max(0, maxAttempts - failedAttempts)
    }

    func recordFailedAttempt() {
        failedAttempts += 1
        if failedAttempts >= maxAttempts {
            lockoutExpiry = Date().addingTimeInterval(900) // 15 minutes
        }
    }

    func resetAttempts() {
        failedAttempts = 0
        lockoutExpiry = Date.distantPast
    }
}
